File: backend/LimsAuth.Api/Models/Lab.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace LimsAuth.Api.Models;

/// <summary>
/// 实验室实体
/// </summary>
[Table("labs")]
public class Lab
{
    [Key]
    [Column("id")]
    public Guid Id { get; set; } = Guid.NewGuid();

    [Required]
    [Column("code")]
    [MaxLength(50)]
    public string Code { get; set; } = string.Empty;

    [Required]
    [Column("name")]
    [MaxLength(200)]
    public string Name { get; set; } = string.Empty;

    /// <summary>
    /// 所属部门ID
    /// </summary>
    [Column("department_id")]
    public Guid? DepartmentId { get; set; }

    /// <summary>
    /// 实验室地点
    /// </summary>
    [Column("location")]
    [MaxLength(200)]
    public string? Location { get; set; }

    /// <summary>
    /// 容纳人数
    /// </summary>
    [Column("capacity")]
    public int Capacity { get; set; } = 0;

    /// <summary>
    /// 实验室类型
    /// </summary>
    [Column("lab_type")]
    [MaxLength(50)]
    public string LabType { get; set; } = "普通实验室";

    /// <summary>
    /// 安全等级
    /// </summary>
    [Column("safety_level")]
    [MaxLength(20)]
    public string SafetyLevel { get; set; } = "一般";

    /// <summary>
    /// 负责人ID
    /// </summary>
    [Column("manager_id")]
    public Guid? ManagerId { get; set; }

    [Column("description")]
    [MaxLength(1000)]
    public string? Description { get; set; }

    [Column("is_active")]
    public bool IsActive { get; set; } = true;

    [Column("created_at")]
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    // 导航属性
    public virtual Department? Department { get; set; }
    public virtual User? Manager { get; set; }
    public virtual ICollection<Equipment> Equipments { get; set; } = new List<Equipment>();
}

/// <summary>
/// 设备实体
/// </summary>
[Table("equipments")]
public class Equipment
{
    [Key]
    [Column("id")]
    public Guid Id { get; set; } = Guid.NewGuid();

    [Required]
    [Column("code")]
    [MaxLength(50)]
    public string Code { get; set; } = string.Empty;

    [Required]
    [Column("name")]
    [MaxLength(200)]
    public string Name { get; set; } = string.Empty;

    /// <summary>
    /// 设备型号
    /// </summary>
    [Column("model")]
    [MaxLength(100)]
    public string? Model { get; set; }

    /// <summary>
    /// 制造商
    /// </summary>
    [Column("manufacturer")]
    [MaxLength(200)]
    public string? Manufacturer { get; set; }

    /// <summary>
    /// 序列号
    /// </summary>
    [Column("serial_number")]
    [MaxLength(100)]
    public string? SerialNumber { get; set; }

    /// <summary>
    /// 所属实验室ID
    /// </summary>
    [Column("lab_id")]
    public Guid? LabId { get; set; }

    /// <summary>
    /// 设备分类
    /// </summary>
    [Column("category")]
    [MaxLength(50)]
    public string Category { get; set; } = "通用设备";

    /// <summary>
    /// 设备状态
    /// </summary>
    [Column("status")]
    [MaxLength(20)]
    public string Status { get; set; } = "正常"; // 正常, 维修中, 报废, 借用中

    /// <summary>
    /// 购买日期
    /// </summary>
    [Column("purchase_date")]
    public DateTime? PurchaseDate { get; set; }

    /// <summary>
    /// 保修期(月)
    /// </summary>
    [Column("warranty_months")]
    public int? WarrantyMonths { get; set; }

    /// <summary>
    /// 购买价格
    /// </summary>
    [Column("price")]
    public decimal? Price { get; set; }

    /// <summary>
    /// 存放位置
    /// </summary>
    [Column("location")]
    [MaxLength(200)]
    public string? Location { get; set; }

    /// <summary>
    /// 设备图片URL
    /// </summary>
    [Column("image_url")]
    [MaxLength(500)]
    public string? ImageUrl { get; set; }

    /// <summary>
    /// 使用说明
    /// </summary>
    [Column("instructions")]
    public string? Instructions { get; set; }

    /// <summary>
    /// 是否需要预约
    /// </summary>
    [Column("requires_booking")]
    public bool RequiresBooking { get; set; } = false;

    /// <summary>
    /// 最大预约时长(小时)
    /// </summary>
    [Column("max_booking_hours")]
    public int? MaxBookingHours { get; set; }

    [Column("description")]
    [MaxLength(1000)]
    public string? Description { get; set; }

    [Column("is_active")]
    public bool IsActive { get; set; } = true;

    [Column("created_at")]
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    // 导航属性
    public virtual Lab? Lab { get; set; }
}

File: frontend/src/api/lab.ts
import axios from 'axios'

const API_BASE_URL = '/api/v1'

// 实验室管理 API
export const labApi = {
  getList: (params?: { keyword?: string; departmentId?: string; labType?: string }) =>
    axios.get(`${API_BASE_URL}/labs`, { params }),
  getById: (id: string) =>
    axios.get(`${API_BASE_URL}/labs/${id}`),
  create: (data: CreateLabRequest) =>
    axios.post(`${API_BASE_URL}/labs`, data),
  update: (id: string, data: UpdateLabRequest) =>
    axios.put(`${API_BASE_URL}/labs/${id}`, data),
  delete: (id: string) =>
    axios.delete(`${API_BASE_URL}/labs/${id}`),
  toggleStatus: (id: string, isActive: boolean) =>
    axios.patch(`${API_BASE_URL}/labs/${id}/status`, { isActive })
}

// 设备管理 API
export const equipmentApi = {
  getList: (params?: { keyword?: string; labId?: string; category?: string; status?: string }) =>
    axios.get(`${API_BASE_URL}/equipments`, { params }),
  getById: (id: string) =>
    axios.get(`${API_BASE_URL}/equipments/${id}`),
  create: (data: CreateEquipmentRequest) =>
    axios.post(`${API_BASE_URL}/equipments`, data),
  update: (id: string, data: UpdateEquipmentRequest) =>
    axios.put(`${API_BASE_URL}/equipments/${id}`, data),
  delete: (id: string) =>
    axios.delete(`${API_BASE_URL}/equipments/${id}`),
  toggleStatus: (id: string, isActive: boolean) =>
    axios.patch(`${API_BASE_URL}/equipments/${id}/status`, { isActive }),
  updateEquipmentStatus: (id: string, status: string) =>
    axios.patch(`${API_BASE_URL}/equipments/${id}/equipment-status`, { status })
}

// 类型定义
export interface CreateLabRequest {
  code: string
  name: string
  departmentId?: string
  location?: string
  capacity?: number
  labType?: string
  safetyLevel?: string
  managerId?: string
  description?: string
}

export interface UpdateLabRequest extends Partial<CreateLabRequest> {
  isActive?: boolean
}

export interface CreateEquipmentRequest {
  code: string
  name: string
  model?: string
  manufacturer?: string
  serialNumber?: string
  labId?: string
  category?: string
  status?: string
  purchaseDate?: string
  warrantyMonths?: number
  price?: number
  location?: string
  imageUrl?: string
  instructions?: string
  requiresBooking?: boolean
  maxBookingHours?: number
  description?: string
}

export interface UpdateEquipmentRequest extends Partial<CreateEquipmentRequest> {
  isActive?: boolean
}

// DTO 类型
export interface LabDto {
  id: string
  code: string
  name: string
  departmentId?: string
  departmentName?: string
  location?: string
  capacity: number
  labType: string
  safetyLevel: string
  managerId?: string
  managerName?: string
  description?: string
  isActive: boolean
  createdAt: string
  equipmentCount: number
}

export interface EquipmentDto {
  id: string
  code: string
  name: string
  model?: string
  manufacturer?: string
  serialNumber?: string
  labId?: string
  labName?: string
  category: string
  status: string
  purchaseDate?: string
  warrantyMonths?: number
  price?: number
  location?: string
  imageUrl?: string
  instructions?: string
  requiresBooking: boolean
  maxBookingHours?: number
  description?: string
  isActive: boolean
  createdAt: string
}

// 常量
export const LAB_TYPES = ['普通实验室', '计算机实验室', '化学实验室', '物理实验室', '生物实验室', '专业实验室']
export const SAFETY_LEVELS = ['一般', '中等', '高危']
export const EQUIPMENT_CATEGORIES = ['通用设备', '计算机设备', '实验仪器', '测量设备', '办公设备', '安全设备', '其他']
export const EQUIPMENT_STATUSES = ['正常', '维修中', '报废', '借用中', '闲置']
